lib/browser/api/app.js

```diff
@@ -11,12 +11,28 @@ const {deprecate, Menu} = electron
 const {EventEmitter} = require('events')
 
 let dockMenu = null
+let readyPromise = null
 
 // App is an EventEmitter.
 Object.setPrototypeOf(App.prototype, EventEmitter.prototype)
 EventEmitter.call(app)
 
 Object.assign(app, {
+  whenReady () {
+    if (readyPromise !== null) {
+      return readyPromise
+    }
+
+    if (app.isReady()) {
+      readyPromise = Promise.resolve()
+    } else {
+      readyPromise = new Promise(resolve => {
+        app.once('ready', resolve)
```

Comment:
This will leak a _tiny_ amount of memory, should switch to `app.once('ready', () => resolve())` to ensure the `launchInfo` isn't left hanging around in this promise that will never be GC'ed.

Reply:
Done.